UefiCpuPkg/SecCore/SecMain.c

```diff
@@ -634,18 +644,57 @@ SecTemporaryRamDone (
     //
     ASSERT (sizeof (UINTN) == sizeof (UINT64));
 
-    Status = MigratePageTable ();
+    //
+    // Initialize page table creation info.
+    //
+    Status = InitPageTableCreationInfo (&Buffer, &BufferSize, &PagingMode, &MaxAddressBits);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, "SecTemporaryRamDone: Failed to initialize page table info: %r.\n", Status));
+      CpuDeadLoop ();
+    }
+
+    //
+    // Create initial range page table defined PcdPageTableInPermanentRamInitSize before Temporary RAM
+    // disabled and CPU cache is not enable for permanent ram.
+    //
+    if (FixedPcdGet64 (PcdPageTableInPermanentRamInitSize) == MAX_UINT64) {
+      Length = LShiftU64 (1, MaxAddressBits);
+    } else {
+      ASSERT (FixedPcdGet64 (PcdPageTableInPermanentRamInitSize) <= LShiftU64 (1, MaxAddressBits));
+      Length = FixedPcdGet64 (PcdPageTableInPermanentRamInitSize);
+    }

```

Comment:
How about?
  Length = FixedPcdGet64 (PcdPageTableInPermanentRamInitSize);
  Length = MAX (LShiftU64 (1, MaxAddressBits), Length);

Reply:
should be:
Length = MIN (LShiftU64 (1, MaxAddressBits), Length);